src/components/Modals/ModalPost/ModalPost.tsx

```diff
@@ -1,72 +1,149 @@
+
+type ModalPostProps = {
+  postId: string;
+  onClose: () => void;
+  newPost?: {
+    text: string;
+    photoUrl: string;
+    authors: {
+      host: {
+        _id: string;
+        name: string;
+        avatar: string;
+      };
+      resident: {
+        _id: string;
+        name: string;
+        avatar: string;
+        species: string;
+      };
+    };
+    createdAt: string;
+    likes: [];
+  };
+};
+
+function ModalPost({ postId, onClose, newPost }: ModalPostProps) {
   const navigate = useNavigate();
-  const dispatch = useDispatch();
-  const currentUser = useSelector(getUser);
-  const modalIsActive = useSelector(getModal);
+  const dispatch = useAppDispatch();
+  const user = useAppSelector(getUser);
+  const posts = useAppSelector(getPosts);
+  const modal = useAppSelector(getModal);
+  const [postText, setPostText] = useState('');
   const [postModalMode, setPostModalMode] = useState(
-    modalIsActive === 'add-post-next' ? 'edit' : 'view'
+    modal.modalIsActive === 'add-post-next' ? 'edit' : 'view'
   );
-  const [postText, setPostText] = useState(text);
+
+  const post = newPost || posts.find((post) => post._id === postId);
+
+  useEffect(() => {
+    if (post) {
+      setPostText(post.text);
+    }
+  }, [post]);
+
+  if (!post) {
+    return <p>Post not found</p>;
+  }
+
+  const { photoUrl, authors, likes, createdAt } = post;
+
+  const postingTime = formatTime(new Date(createdAt));
+  const isToday =
+    createdAt.slice(0, 10) === new Date().toISOString().slice(0, 10);
 
   const handleEditClick = () => {
-    console.log('edit');
     setPostModalMode('edit');
   };
 
   const handleTextChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
     setPostText(e.target.value);
   };
 
+  const handleDeleteClick = () => {
+    deletePosts(localStorage.jwt, postId).then(() => {
+      dispatch(removePost({ _id: postId }));
+      onClose();
+    });
+  };
+
   const handleSaveClick = () => {
-    const create = () => {
-      const newPost = {
-        id: Date.now(), // temp ID
-        text: postText,
-        photoUrl: photoUrl,
-        authors: authors,
-        createdAt: new Date().toISOString(),
-        likes: 0,
-      };
-      dispatch(addPost(newPost));
-      navigate('/');
+    const newPost = {
+      text: postText,
+      photoUrl: photoUrl,
+      residentId: authors.resident._id,
     };
-    const update = () => {
-      dispatch(updatePost({ id, newText: postText }));
-      console.log('save post');
+
+    const updatedPost = {
+      id: postId,
+      text: postText,
+    };
+
+    const closeAfterSaving = () => {
+      setPostModalMode('view');
+      onClose();
     };
-    modalIsActive === 'add-post-next' ? create() : update();
-    setPostModalMode('view');
-    onClose();
+
+    modal.modalIsActive === 'add-post-next'
+      ? saveNewPost(newPost).then(() => closeAfterSaving())
+      : saveUpdatedPost(updatedPost).then(() => closeAfterSaving());
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**